TradeItIosTicketSDK2/TradeItDeviceManager.swift

```diff
@@ -0,0 +1,69 @@
+import LocalAuthentication
+
+class TradeItDeviceManager {
+    
+    func authenticateUserWithTouchId(onSuccess onSuccess: () -> Void, onFailure: () -> Void) {
+        var error: NSError?
+        let myLocalizedReasonString = "Authentication is required"
+        
+        let context = LAContext()
+        
+        guard context.canEvaluatePolicy(LAPolicy.DeviceOwnerAuthentication, error: &error) else {
+            print("touch id not available on this device: \(error)")
+            onSuccess()
+            return
+        }
+        
+        context.evaluatePolicy(LAPolicy.DeviceOwnerAuthentication,
+                               localizedReason: myLocalizedReasonString,
+                               reply: { (success: Bool, evaluationError: NSError?) -> Void in
+                                
+                                if success {
+                                    print("authenticateUserWithTouchId succeeded")
+                                    onSuccess()
+                                }
+                                else {
+                                    guard let error = evaluationError else {
+                                        print("authenticateUserWithTouchId failed")
+                                        onFailure()
+                                        return
+                                    }
+                                    
+                                    switch error.code {
+                                    case LAError.AuthenticationFailed.rawValue:
+                                        print("Authentication failed: invalid credentials")
+                                        onFailure()
+                                    case LAError.UserCancel.rawValue:
+                                        print("Authentication cancelled by the user")
+                                        onFailure()
+                                    case LAError.UserFallback.rawValue:
+                                        print("User tapped the fallback button")
+                                        onFailure()
+                                    case LAError.SystemCancel.rawValue:
+                                        print("Canceled by system (another application went to foreground)")
+                                        onFailure()
+                                    case LAError.PasscodeNotSet.rawValue:
+                                        print("Passcode not set")
+                                        onFailure()
+                                    case LAError.TouchIDNotAvailable.rawValue:
+                                        print("Touch id not available on the device")
+                                        onFailure()
```

Comment:
I think it will be catch by the first guard, so we shouldn't have this case (I tested it on the test device). But correct, I am going to fix it.